src/angular/11-declarative-state/mergeMap.html

```diff
@@ -0,0 +1,47 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/rxjs/6.3.3/rxjs.umd.min.js"></script>
+<script type="module">
+  const { of, interval, zip } = rxjs;
+  const { map, mergeMap, delay } = rxjs.operators;
+
+  // returns a stream that emits the price of a given car after a 500ms delay. this is how a request might operate
+  function pseudoPriceRequest(carName) {
+    if (carName === 'mustang') {
+      return of(40000).pipe(delay(500));
+    } else if (carName === 'camaro') {
+      return of(38000).pipe(delay(500));
+    }
+  }
+
+  // emits the values every 2 seconds in order
+  const pseudoFormValueStream$ = zip(
+    of('', 'mustang', 'camaro', ''),
+    interval(2000)
+  ).pipe(map(([val, num]) => val));
+
+  // stream that makes a "request" if provided a car name or returns "No Car Selected"
+  const pseudoPriceStream$ = pseudoFormValueStream$.pipe(
+    mergeMap((selectedCar) => {
+      if (selectedCar) {
```

Comment:
or instead do something like:
```typescript
const priceRequest$ = pseudoPriceRequest(selectedCar);

if (!priceRequest$) { throw new Error("Unexpected missing price request") }
```

Reply:
since we are opening new issue https://bitovi.atlassian.net/browse/NGACAD-80 for changing the example to a shopping cart example, I think this could be done there.